pkg/controllers/cspi-controller/handler.go

```diff
@@ -346,3 +355,16 @@ func (c *CStorPoolInstanceController) addPoolProtectionFinalizer(
 		string(cspi.GetUID()))
 	return newCSPI, nil
 }
+
+func (c *CStorPoolInstanceController) sync(cspi *cstor.CStorPoolInstance) {
+	// Right now the only sync activity is compression
+	compressionType := cspi.Spec.PoolConfig.Compression
+	poolName := zpool.PoolName()
+	err := zpool.SetCompression(poolName, compressionType)
```

Comment:
Before setting we should do get if it doesn't match then good to set. Instead of setting for every reconciliation.

Reply:
Not required IMO unless there are any specific reasons.

Reply:
Here we are performing a write operation on the disk for every 30 sec which is not good.

Reply:
Done.